pennylane/templates/subroutines/qsvt.py

```diff
@@ -0,0 +1,193 @@
+            def example_circuit(A):
+                qml.qsvt(A, lst_phis, wires=[0,1])
+                return qml.expval(qml.PauliZ(wires=0))
+
+            qml.draw(example_circuit)(A)
+    """
+    c, r = qml.math.shape(A)
+
+    U_A = qml.BlockEncode(A, wires=wires, do_queue=False)
+    lst_projectors = []
+    for idx, phi in enumerate(phi_vect):
+        if idx % 2 == 0:
+            lst_projectors.append(PCPhase(phi, dim=r, wires=wires, do_queue=False))
+        else:
+            lst_projectors.append(PCPhase(phi, dim=c, wires=wires, do_queue=False))
+
+    return QSVT(U_A, lst_projectors, wires=wires)
+
+
+class QSVT(Operation):
+    r"""Performs the 
+    `quantum singular value transformation <https://arxiv.org/abs/1806.01838>`__ circuit.
+
+    Given a circuit :math:`U(A)`, which block encodes the matrix :math:`A`, and a list of projector-controlled
+    phase shifts, this template applies the circuit for quantum singular value transformation.
+
+    .. math::
+
+        \begin{align}
+             U_{qsvt}(A, \phi) &=
+             \begin{bmatrix}
+                Poly^{SV}(A) & \cdot \\
+                \cdot & \cdot
+            \end{bmatrix}.
+        \end{align}
+
+    This circuit can be used to perform the standard quantum singular value transformation algorithm, consisting
+    of alternating block encoding and controlled phase shift operations.
+
+    Args:
+        U_A (Operator): the block encoding circuit, specified as a :class:`~.Operator` 
+            or quantum function.
+        lst_projectors (list[Operator] or list[Callable]): a list of projector-controlled phase
+            shifts that implement the desired polynomial.
+        wires (Iterable): the wires the template acts on.
+
+    Raises:
+        QuantumFunctionError: 
+
+    .. details::
+        :title: Usage Details
+
+        Consider the matrix corresponding to a rotation from an :class:`~.RX` gate:
+
+        .. code-block:: python
+
+            dev = qml.device('default.qubit', wires=2)
+            A = [[0.1]]
+            blckencode = qml.BlockEncode(A,wires=[0,1])
+            lst_phis = [qml.PCPhase(i+0.1,dim=1, wires=[0,1]) for i in range(3)]
+            @qml.qnode(dev)
+            def example_circuit(A):
+                qml.QSVT(blckencode,lst_phis,wires=[0,1])
+                return qml.expval(qml.PauliZ(wires=0))
+                
+            qml.matrix(example_circuit)(A)
+
+            
+        Continue any explanation here.
+    """
+
+    num_wires = AnyWires
+    """int: Number of wires that the operator acts on."""
+
+    # ndim_params = (0, 1,)
+    # """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    grad_method = None
+    """Gradient computation method."""
+
+    def __init__(self, U_A, lst_projectors, wires, do_queue=True, id=None):
+        if not isinstance(U_A, qml.operation.Operator):
+            raise ValueError("Input block encoding must be an Operator")
```

Comment:
It can be a quantum function also?

Reply:
Not currently. I'm working to add this in.

Reply:
What is the plan for passing the parameters of the callable? We may want to skip qfunc support for now.

Reply:
Decided to skip qfunc support for now. Users have to input operators.